internal/task/task.go

```diff
@@ -50,31 +47,42 @@ func New(cfg config.Config) (*Task, error) {
 // The container vCPU can not be greater than Task CPU limit, therefore if
 // Task CPU limit is less than 1, the max threads returned is 1.
 // If no CPU limit is found for the container, then the max number of threads
-// returned is the number of CPU's for the ECS Task. If no CPU limit is found for the
-// Task, then 0 is returned with an error
+// returned is the number of CPU's for the ECS Task.
 func (t *Task) GetMaxProcs() (int, error) {
+	container, err := t.getContainerMeta()
+	if err != nil {
+		return 0, fmt.Errorf("failed to get ECS container meta: %w", err)
+	}
+
 	task, err := t.getTaskMeta()
 	if err != nil {
-		return 0, fmt.Errorf("failed to get ECS task: %w", err)
+		return 0, fmt.Errorf("failed to get ECS task meta: %w", err)
 	}
 
-	if task.Limits.CPU == 0 {
-		return 0, fmt.Errorf("no CPU limit found for task")
+	// Either the container limit or the task limit must be set
+	if container.Limits.CPU == 0 && task.Limits.CPU == 0 {
+		return 0, fmt.Errorf("no CPU limit found for task or container")
 	}
 
-	var cpuLimit float64
-	for _, container := range task.Containers {
-		if container.DockerID == t.cfg.ConainerID {
-			cpuLimit = container.Limits.CPU
+	var containerCpuLimit float64
+	for _, taskContainer := range task.Containers {
+		if container.DockerID == taskContainer.DockerID {
+			containerCpuLimit = taskContainer.Limits.CPU
 		}
 	}
 
-	if cpuLimit == 0 {
+	if containerCpuLimit == 0 {
 		minThreads := 1
 		return max(int(task.Limits.CPU), minThreads), nil
 	}
 
-	cpu := int(cpuLimit) >> 10
+	cpu := int(containerCpuLimit) >> 10
+	taskCpuLimit := int(task.Limits.CPU)
+	if taskCpuLimit > 0 {
+		if cpu > taskCpuLimit {
+			return int(taskCpuLimit), nil
+		}
+	}
```

Comment:
**nit** if possible can we avoid double nested if statements and do something along the lines of the below. Also I can see we are casting taskCpuLimit here to an int on line 83 when we have already casted to an int on line 80, so I think we might be sweet here to remove the casting on line 83. What you think?
```
if taskCpuLimit > 0 {
               return min(cpu, taskCPULimit)
	}
```